### DISCUSSION TITLE
Observer pattern and queued listeners

### DISCUSSION THREAD

**Discussion by opicodes** [laravel-zero/laravel-zero#397] [Q&A]
I've implemented the observer pattern in my project by adding an Event and a Listener and created an `EventServiceProvider` like so:

```
class EventServiceProvider extends ServiceProvider
{
    protected $listen = [
        WspNotificationsReceived::class => [ProcessWspNotifications::class]
    ];
    ...
```

which I registered in `config/app.php` like so:

```
    'providers' => [
        AppServiceProvider::class,
        EventServiceProvider::class
    ],
```

This should be working right?

Now the real question. 

In Laravel you can queue your listeners implementing the `Illuminate\Contracts\Queue\ShouldQueue` contract and defining a `queue` property on the Listener class `public string $queue = 'listeners'` as described [here](https://laravel.com/docs/9.x/events#queued-event-listeners). 

Now, although the ShouldQueue contract is available in Laravel Zero, it seems to me that everything else queue related is missing, starting with artisan commands. Am I right? If so, is it something that could be added somehow?

Thanks

**Comment by owenvoke** [comment]
Does using the Queue component help with your situation? (i.e. `php <your-app> app:install queue`). Interestingly it doesn't look like this component is actually documented. 🤔

**Reply by opicodes** [reply]
Can I do that? I assumed I couldn't as it's not in the docs :)

Sorry for being such a newbie, but I come from symfony where almost everything is a component and can be used "standalone".
Thanks again

**Reply by owenvoke** [reply]
You should be able to... Basically if you run `php <your-app> app:install` it will show a list of available components you can install.
